server/sockets/list.js

```diff
@@ -517,6 +518,50 @@ const removeListsOnArchiveCohort = (socket, dashboardClients) =>
       });
   });
 
+const emitListsOnRestoreCohort = (socket, dashboardClients, cohortClients) =>
+  socket.on(CohortActionTypes.RESTORE_SUCCESS, data => {
+    const { cohortId } = data;
+
+    List.find({ cohortId })
+      .populate('cohortId', 'ownerIds')
+      .lean()
+      .exec()
+      .then(docs => {
+        if (docs) {
+          const {
+            cohortId: { ownerIds: cohortOwners }
+          } = docs[0];
+          const listsByUsers = getListsDataByViewers(docs);
+
+          cohortOwners.forEach(id => {
+            const cohortOwnerId = id.toString();
+
+            if (cohortClients.has(cohortOwnerId)) {
+              const { socketId, viewId } = cohortClients.get(cohortOwnerId);
+              if (viewId === cohortId) {
```

Comment:
is this check necessary, doesn't the line 539 claims that owner is currently on that archived cohort view?

Reply:
It is necessary because if the owner of cohort "A" will be on cohort "B" view their id will be stored in `cohortClients` also. But there is no need to send the list for them.